app/v2/controllers/V2MovementsController.scala

```diff
@@ -0,0 +1,328 @@
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.play.http.HeaderCarrierConverter
+import v2.controllers.actions.AuthNewEnrolmentOnlyAction
+import v2.controllers.actions.providers.AcceptHeaderActionProvider
+import v2.controllers.actions.providers.MessageSizeActionProvider
+import v2.controllers.request.AuthenticatedRequest
+import v2.controllers.stream.StreamingParsers
+import v2.models.AuditType
+import v2.models.EORINumber
+import v2.models.MessageId
+import v2.models.MovementId
+import v2.models.MovementType
+import v2.models.errors.PresentationError
+import v2.models.request.MessageType
+import v2.models.responses.MovementResponse
+import v2.models.responses.UpdateMovementResponse
+import v2.models.responses.hateoas._
+import v2.services._
+import v2.utils.PreMaterialisedFutureProvider
+
+import java.time.OffsetDateTime
+import scala.concurrent.Future
+
+@ImplementedBy(classOf[V2MovementsControllerImpl])
+trait V2MovementsController {
+  def createMovement(movementType: MovementType): Action[Source[ByteString, _]]
+  def getMessage(movementType: MovementType, movementId: MovementId, messageId: MessageId): Action[AnyContent]
+  def getMessageIds(movementType: MovementType, movementId: MovementId, receivedSince: Option[OffsetDateTime] = None): Action[AnyContent]
+  def getMovement(movementType: MovementType, movementId: MovementId): Action[AnyContent]
+  def getMovements(movementType: MovementType, updatedSince: Option[OffsetDateTime]): Action[AnyContent]
+  def attachMessage(movementType: MovementType, movementId: MovementId): Action[Source[ByteString, _]]
+}
+
+@Singleton
+class V2MovementsControllerImpl @Inject() (
+  val controllerComponents: ControllerComponents,
+  authActionNewEnrolmentOnly: AuthNewEnrolmentOnlyAction,
+  validationService: ValidationService,
+  conversionService: ConversionService,
+  movementsService: MovementsService,
+  routerService: RouterService,
+  auditService: AuditingService,
+  pushNotificationsService: PushNotificationsService,
+  messageSizeAction: MessageSizeActionProvider,
+  acceptHeaderActionProvider: AcceptHeaderActionProvider,
+  val metrics: Metrics,
+  xmlParsingService: XmlMessageParsingService,
+  jsonParsingService: JsonMessageParsingService,
+  responseFormatterService: ResponseFormatterService,
+  val preMaterialisedFutureProvider: PreMaterialisedFutureProvider
+)(implicit val materializer: Materializer, val temporaryFileCreator: TemporaryFileCreator)
+    extends BaseController
+    with V2MovementsController
+    with Logging
+    with StreamingParsers
+    with VersionedRouting
+    with ErrorTranslator
+    with ContentTypeRouting
+    with HasActionMetrics {
+
+  lazy val sCounter: Counter = counter(s"success-counter")
+  lazy val fCounter: Counter = counter(s"failure-counter")
+
+  def createMovement(movementType: MovementType): Action[Source[ByteString, _]] =
+    movementType match {
+      case MovementType.Arrival =>
+        contentTypeRoute {
```

Comment:
I think best to pass movementType as parameter with submitArrivalNotificationXML and submitArrivalNotificationJSON and use it later instead of hardcoding every time later in code

Reply:
We need to determine what the movement type is at the router level, for the distinct endpoints

Reply:
yes agree. By passing it to submitArrivalNotificationXML and submitArrivalNotificationJSON we can achieve it.

Reply:
Oh I see what you're saying - but no, it's hard coded because it'll never change from being an arrival and it isn't hard coded from the router because it could be an arrival or departure.

Reply:
Fine. No problem